invenio_records_resources/services/record.py

```diff
@@ -49,52 +49,73 @@ class RecordService(Service):
     """Record Service."""
 
     default_config = RecordServiceConfig
+    # Singleton objects
+    _resolver = None
+    _minter = None
+    _fetcher = None
+    _indexer = None
+    _search_engine = None
 
     #
     # Low-level API
     #
-
-    #
-    # Persistent identifier resolution
-    #
+    @property
```

Comment:
nit: I think is easier to use `functools.cached_property` :) You will avoid all the `_` attributes for singleton...

Reply:
Nice! 🤝